bin/propolis-server/src/lib/stats/virtual_machine.rs

```diff
@@ -0,0 +1,530 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// Copyright 2024 Oxide Computer Company
+
+//! Types for tracking statistics about virtual machine instances.
+
+// Propolis is built in a variety of configurations, including checks and tests
+// run on non-illumos machines where kstats are meaningless. This is a big
+// hammer, but a large number of the values in this module are not referenced in
+// those configurations, and so this is more straightfoward than littering the
+// code with cfg directives.
+#![cfg_attr(any(test, not(target_os = "illumos")), allow(dead_code))]
```

Comment:
As a follow-up item, I'd like to update kstat-rs to be able to stub out its function bindings, so it can be included (albeit in a non-functional matter) on platforms which lack the library.  Not at all blocking for this, though.

Reply:
Yeah, good idea. I filed https://github.com/oxidecomputer/kstat-rs/issues/1 to track this.